web/templates/recipient_row.hbs

```diff
@@ -57,19 +57,28 @@
                 {{/if}}
             {{/if}}
 
-            {{#if topic_muted}}
-                <i class="zulip-icon zulip-icon-mute on_hover_topic_unmute recipient_bar_icon" data-stream-id="{{stream_id}}" data-topic-name="{{topic}}" data-tooltip-template-id="topic-unmute-tooltip-template" role="button" tabindex="0" aria-label="{{t 'Unmute topic' }}"></i>
-                <template id="topic-unmute-tooltip-template">
-                    {{t "Unmute topic" }}
-                    {{tooltip_hotkey_hints "M"}}
-                </template>
+
+            {{#if stream_muted}}
+                {{#if topic_unmuted}}
+                    <i class="zulip-icon zulip-icon-unmute on_hover_topic_toggled recipient_bar_icon hidden-for-spectators" data-stream-id="{{stream_id}}" data-topic-name="{{topic}}" data-tooltip-template-id="topic-mute-tooltip-template" role="button" tabindex="0" aria-label="{{t 'Mute topic' }}"></i>
+                {{else}}
+                    <i class="zulip-icon zulip-icon-unmute on_hover_topic_toggle recipient_bar_icon" data-stream-id="{{stream_id}}" data-topic-name="{{topic}}" data-tooltip-template-id="topic-unmute-tooltip-template" role="button" tabindex="0" aria-label="{{t 'Unmute topic' }}"></i>
+                {{/if}}
             {{else}}
-                <i class="zulip-icon zulip-icon-mute on_hover_topic_mute recipient_bar_icon hidden-for-spectators" data-stream-id="{{stream_id}}" data-topic-name="{{topic}}" data-tooltip-template-id="topic-mute-tooltip-template" role="button" tabindex="0" aria-label="{{t 'Mute topic' }}"></i>
-                <template id="topic-mute-tooltip-template">
-                    {{t "Mute topic" }}
-                    {{tooltip_hotkey_hints "M"}}
-                </template>
+                {{#if topic_muted}}
+                    <i class="zulip-icon zulip-icon-mute on_hover_topic_toggled recipient_bar_icon" data-stream-id="{{stream_id}}" data-topic-name="{{topic}}" data-tooltip-template-id="topic-unmute-tooltip-template" role="button" tabindex="0" aria-label="{{t 'Unmute topic' }}"></i>
+                {{else}}
+                    <i class="zulip-icon zulip-icon-mute on_hover_topic_toggle recipient_bar_icon hidden-for-spectators" data-stream-id="{{stream_id}}" data-topic-name="{{topic}}" data-tooltip-template-id="topic-mute-tooltip-template" role="button" tabindex="0" aria-label="{{t 'Mute topic' }}"></i>
+                {{/if}}
             {{/if}}
```

Comment:
We can condense it by using `topic_unmuted` and `topic_muted` conditions inside the `class=`.